units/en/unit7/self-play.mdx

```diff
@@ -0,0 +1,137 @@
+
+The Elo system is **inferred from the losses and draws against other players.** It means that player ratings depend **on the ratings of their opponents and the results scored against them.**
+
+Elo defines an Elo score that is the relative skills of a player in a zero-sum game. **We say relative because it depends on the performance of opponents.**
+
+The central idea is to think of the performance of a player **as a random variable that is normally distributed.**
+
+The difference in rating between 2 players serves as **the predictor of the outcomes of a match.** If the player wins but the probability is high it will not win a lot of points from their opponent, since it means that it was much stronger than it.
+
+After every game:
+
+- The winning player takes **points from the losing one.**
+- The number of points is determined **by the difference in the 2 players ratings (hence relative).**
+    - If the higher-rated player wins → few points will be taken from the lower-rated player.
+    - If the lower-rated player wins → a lot of points will be taken from the high-rated player.
+    - If it’s a draw → the lower-rated player gains a few points from the higher.
+
+So if A and B have rating Ra, and Rb, then the **expected scores are** given by:
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit10/elo1.png" alt="ELO Score"/>
+
+Then, at the end of the game, we need to update the player’s actual Elo score, we use a linear adjustment **proportional to the amount by which the player over-performed or under-performed.**
+
+We also define a maximum adjustment rating per game: K-factor.
+
+- K=16 for master.
+- K=32 for weaker players.
+
+If Player A has Ea points but scored Sa points, then the player’s rating is updated using the formula:
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/unit10/elo2.png" alt="ELO Score"/>
+
+### Example
+
+If we take an example:
+
+Player A has a rating of 2600
+
+Player B has a rating of 2300
+
+- We first calculate the expected score:
+
+\\(E_{A} = \frac{1}{1+10^{(2300-2600)/400}} = 0.849 \\)
+
+\\(E_{B} = \frac{1}{1+10^{(2600-2300)/400}} = 0.151 \\)
+
+- If the organizers determined that K=16 and A wins, the new rating would be:
+
+\\(ELO_A = 2600 + 16*(1-0.849) = 2602 \\)
+
+\\(ELO_B = 2300 + 16*(1-0.151) = 2298 \\)
+
+- If the organizers determined that K=16 and B wins, the new rating would be:
+
+\\(ELO_A = 2600 + 16*(0-0.849) = 2586 \\)
+
+\\(ELO_B = 2300 + 16 *(1-0.151) = 2314 \\)
+
+
+### The Advantages
+
+Using ELO score has multiple advantages:
+
+- Points are **always balanced** (more points are exchanged when there is an unexpected outcome but the sum is always the same).
+- It is a **self-corrected system** since if a player wins against a weak player, you will not win a lot of points.
+- If **works with team games**: we calculate the average for each team and use it in Elo.
```

Comment:
```suggestion
- Points are **always balanced** (more points are exchanged when there is an unexpected outcome, but the sum is always the same).
- It is a **self-corrected system** since if a player wins against a weak player, you will only win a few points.
- If **works with team games**: we calculate the average for each team and use it in Elo.
```